### PR TITLE
Better pfs error msgs

### PR DESCRIPTION
**Pull request by hackaugusto**
Extended routing error messages

    Now the node will always validate the transfer before doing a PFS
    request. If the node does not have channels that can even start the
    transfers, then an error is reported and the PFS is not requested, this
    includes:

    - If there are no channels open
    - If there are no channels with enough distributable to start to mediate
      the transfer (this assumes transfers are not split)
    - If there is no route to the target with any of the existing channels
      (this assumes the route will always be in the same token network).

### CODE REVIEW

File: raiden/routing.py [hunk 1]
```diff
@@ -18,16 +18,25 @@
...
+    PaymentWithFeeAmount,
     TargetAddress,
     TokenNetworkAddress,
+    Tuple,
 )
 
 log = structlog.get_logger(__name__)
 
 
+class ErrorDirectChannel(Enum):
```
Comment (palango) [inline raiden/routing.py]: nit: I feel `DirectChannelError` would be the better name.

File: raiden/transfer/channel.py [hunk 2]
```diff
@@ -133,6 +134,16 @@ class UnlockGain(NamedTuple):
     from_partner_locks: TokenAmount
 
 
+class IsChannelUsable(Enum):
```
Comment (palango) [inline raiden/transfer/channel.py]: This sounds to me like a boolean, what about `ChannelUsability` and renaming `YES` to `USABLE`?

File: raiden/transfer/mediated_transfer/initiator.py [hunk 3]
```diff
@@ -267,7 +267,7 @@ def try_new_route(
             transfer_amount=amount_with_fee,
             lock_timeout=transfer_description.lock_timeout,
         )
-        if is_channel_usable:
```
Comment (palango) [inline raiden/transfer/mediated_transfer/initiator.py]: I's also rename this variable, this sounds too boolean for my taste.

File: raiden/transfer/channel.py [hunk 4]
```diff
@@ -209,14 +220,14 @@ def is_channel_usable_for_mediation(
...
-    return channel_usable
+    return channel_usable is IsChannelUsable.YES
 
 
 def is_channel_usable_for_new_transfer(
     channel_state: NettingChannelState,
     transfer_amount: PaymentWithFeeAmount,
     lock_timeout: Optional[BlockTimeout],
-) -> bool:
+) -> IsChannelUsable:
```
Comment (palango) [inline raiden/transfer/channel.py]: Nice, this makes the failure clearer!

File: raiden/routing.py [hunk 5]
```diff
@@ -40,47 +49,124 @@ def get_best_routes(
...
-            to_address=to_address,
-            amount=amount,
-            previous_address=previous_address,
+    token_network = views.get_token_network_by_address(chain_state, token_network_address)
+    assert token_network, "The token network must be validated and exist."
+
+    try:
+        # networkx returns a generator, consume the result since it will be
+        # iterated over multiple times.
+        all_neighbors = list(
```
Comment (palango) [inline raiden/routing.py]: Any particular reason for using this over `views.all_neighbour_nodes()`?

File: raiden/routing.py [hunk 6]
```diff
@@ -40,47 +49,124 @@ def get_best_routes(
...
+    #   available.
+    # - There will be no mediation fees
+    # - The transfer will be faster
+    if to_address in all_neighbors:
+        for channel_id in token_network.partneraddresses_to_channelidentifiers[
+            Address(to_address)
+        ]:
+            channel_state = token_network.channelidentifiers_to_channels[channel_id]
+
+            # direct channels dont have fees
```
Comment (palango) [inline raiden/routing.py]: s/dont/don't

File: raiden/routing.py [hunk 7]
```diff
@@ -40,47 +49,124 @@ def get_best_routes(
...
+                error_no_route += 1
+            else:
+                distributable = channel.get_distributable(
+                    channel_state.our_state, channel_state.partner_state
+                )
+
+                if distributable < amount:
+                    error_no_capacity += 1
+                else:
+                    nonrefundable = amount > channel.get_distributable(
```
Comment (palango) [inline raiden/routing.py]: Not for this PR, but it feels like we have the same checks all over the place. We should try to pull this out to have it in one place.

File: raiden/routing.py (hunk 5 above)
Comment (hackaugusto) [reply raiden/routing.py]: I didnt want to change the code, just to add the validation

**Pull request merged**